modules/oe_editorial_unpublish/src/UnpublishableContentEntities.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_editorial_unpublish;
+
+use Drupal\content_moderation\ModerationInformation;
+use Drupal\Core\Entity\ContentEntityTypeInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Determines the content entity types that can be unpublished.
+ */
+class UnpublishableContentEntities {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The content moderation information service.
+   *
+   * @var \Drupal\content_moderation\ModerationInformation
+   */
+  protected $moderationInformation;
+
+  /**
+   * UnpublishableContentEntities constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   The entity type manager.
+   * @param \Drupal\content_moderation\ModerationInformation $moderationInformation
+   *   The content moderation information service.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, ModerationInformation $moderationInformation) {
+    $this->entityTypeManager = $entityTypeManager;
+    $this->moderationInformation = $moderationInformation;
+  }
+
+  /**
+   * Returns the content entity types that can be unpublished.
+   *
+   * @return \Drupal\Core\Entity\ContentEntityTypeInterface[]
+   *   The content entity type definitions.
+   */
+  public function getUnpublishableDefinitions(): array {
```

Comment:
For me it would tell more if we would call this getUnpublishableEntityTypes() just like your annotation says. We never really call them defintions and $definition->id() would better fit $entity_type->id(). Up to you.

Reply:
It's correct the way it is.